cmake/make_dist/MakeDistCheck.cmake

```diff
@@ -22,6 +22,11 @@ function (RUN_DIST_CHECK PACKAGE_PREFIX EXT)
       ERROR_MSG "Command to untar ${tarball} failed."
    )
 
+   # Ensure a VERSION_CURRENT file is present.
+   if(NOT EXISTS ${PACKAGE_PREFIX}/build/VERSION_CURRENT)
```

Comment:
This check should be guarded by the same `"${CMAKE_BINARY_DIR}" STREQUAL "${PROJECT_SOURCE_DIR}/build"` condition used to determine if the VERSION_CURRENT file should be included in the tarball. Otherwise, distcheck fails when `${CMAKE_BINARY_DIR}` is not `./build`.

Suggest defining a variable (i.e. `MONGOCXX_INCLUDE_VERSION_FILE_IN_DIST`) to track this condition:

```cmake
set(MONGOCXX_INCLUDE_VERSION_FILE_IN_DIST OFF)
set(MONGOCXX_CURRENT_VERSION_FILE "")

# ...

        if("${CMAKE_BINARY_DIR}" STREQUAL "${PROJECT_SOURCE_DIR}/build")
            set(MONGOCXX_INCLUDE_VERSION_FILE_IN_DIST ON)
            set(MONGOCXX_CURRENT_VERSION_FILE build/VERSION_CURRENT)
        endif ()

# ...
```
```cmake
# Ensure a VERSION_CURRENT file is present.
if(MONGOCXX_INCLUDE_VERSION_FILE_IN_DIST AND "${CMAKE_BINARY_DIR}" STREQUAL "${PROJECT_SOURCE_DIR}/build")
   if(NOT EXISTS ${PACKAGE_PREFIX}/build/VERSION_CURRENT)
      message (FATAL_ERROR "Expected tarball to include `build/VERSION_CURRENT` file but did not.")
   endif ()
endif()
```

Reply:
Fixed suggestion to omit redundant condition.

Reply:
Thank you for the suggestion. Updated as suggested.